### ISSUE TITLE
BUG REPORT

### ISSUE THREAD

**Issue by Patrick-Ni** [princeton-nlp/AutoCompressors#8]
Hi, I'm trying to reproduce this excellent work. I found some suspected bugs in line 232 and 233 of AutoCompressor.py.
When i print two variables: ``soft_prompt_length`` and ``summary_length``, i find that during the ``forward_segment`` loop, these two variables will become 0 at some time before loop ends, which will cause the ``shift_logits`` dimension to be summary_length longer than ``labels`` when calculating loss. I speculate that this may be caused by calling ``SummaryConfig.reset`` early?
Anyway, I chose to replace the code in these two places with:
```
soft_prompt_length = soft_prompt.size(1)
summary_length = placeholder_embeds.size(1)
```
to avoid errors. Can you guys check the reason for this?

**Comment by CodeCreator** [comment]
Thank you for your interest in our work! 

I'm struggling to reproduce the bug, as `reset` is only called at the end of the forward pass. But I also think that your proposed solution is equally correct and makes the statements clearer in those lines. I have gone ahead and commited this change to the repo. 

Please let me know if you run into any other issues!
